### PR TITLE
Always run single instance of tdnf

### PR DESCRIPTION
**Pull request by sshedi**
Signed-off-by: Shreenidhi Shedi <sshedi@vmware.com>

### CODE REVIEW

File: tools/cli/defines.h [hunk 1]
```diff
@@ -21,6 +21,8 @@
 
 #pragma once
 
+#define TDNF_INSTANCE_LOCK_FILE     "/tmp/.tdnf-instace-lockfile"
```
Comment (sidcha) [inline tools/cli/defines.h]: Can we stick to conventions put this inside inside `/var/lock/` or `/var/run/lock/`

File: tools/cli/main.c [hunk 2]
```diff
@@ -63,13 +69,26 @@ int main(int argc, char* argv[])
...
+    /*
+     * granular permissions for non root users are pending.
+     * blocking all operations for non root and show the
+     * right error to avoid confusion
+     */
+    if (geteuid())
     {
-        dwError = ERROR_TDNF_PERM;
-        BAIL_ON_CLI_ERROR(dwError);
+        BAIL_ON_CLI_ERROR((dwError = ERROR_TDNF_PERM));
```
Comment (sidcha) [inline tools/cli/main.c]: This looks little aggressive :). Even if its not as granular as it must be, we must to this at a per-command level at the very least. Some commands like tdnf list should be runnable by any user.

File: tools/cli/main.c [hunk 3]
```diff
@@ -63,13 +69,26 @@ int main(int argc, char* argv[])
...
+    }
+
+    if (IsTdnfAlreadyRunning())
```
Comment (sidcha) [inline tools/cli/main.c:83]: Again, only modify operations needs to take this lock. We must defer this call further down, maybe at per-comand bassis.

**Review by sidcha — CHANGES_REQUESTED** [review]
Some more work is needed. But a step in the good direction. :)

Thanks!

File: tools/cli/main.c (hunk 2 above)
Comment (sshedi) [reply tools/cli/main.c]: This should be a separate enhancement. Let's do it later

**Review by ppadmavilasom — APPROVED** [review]
LGTM

**Review by keerthanakalyan — APPROVED** [review]

**Pull request merged**